contracts/src/AgentExecutor.sol

```diff
@@ -36,4 +50,25 @@ contract AgentExecutor {
     function _transferToken(address token, address recipient, uint128 amount) internal {
         IERC20(token).safeTransfer(recipient, amount);
     }
+
+    /// @dev Register native asset from polkadto as ERC20 `token`.
+    function _registerToken(bytes32 agentID, bytes32 tokenID, string memory name, string memory symbol, uint8 decimals)
+        internal
+    {
+        IERC20 token = new ERC20(name, symbol, decimals);
+        Gateway(msg.sender).registerForeignToken(tokenID, address(token), agentID);
+    }
+
+    /// @dev Mint ERC20 token to `recipient`.
+    function _mintToken(bytes32 tokenID, address recipient, uint256 amount) internal {
+        address token = Gateway(msg.sender).tokenAddressOf(tokenID);
+        ERC20(token).mint(recipient, amount);
+        emit TokenMinted(tokenID, token, recipient, amount);
+    }
+
+    function burnToken(bytes32 tokenID, address sender, uint256 amount) external {
+        address token = Gateway(msg.sender).tokenAddressOf(tokenID);
```

Comment:
Ditto: pass `token` in from Gateway.sol.

Reply:
https://github.com/Snowfork/snowbridge/pull/1155/commits/9b7811d301f04d2417fd9fad3e4ba1bc1dfc59a5